dashboard/src/actions/overviewActions.js

```diff
@@ -57,7 +57,12 @@ const initializeRuns = () => (dispatch, getState) => {
   data.forEach((item) => {
     item[CONSTANTS.IS_EDIT] = false;
     item[CONSTANTS.NAME_COPY] = item.name;
-    item[CONSTANTS.IS_DIRTY] = false;
+    item[CONSTANTS.SERVER_DELETION_COPY] =
+      item.metadata[CONSTANTS.SERVER_DELETION];
+    item[CONSTANTS.IS_DIRTY] = {
+      serverDelete: false,
+      datasetName: false,
+    };
```

Comment:
I don't see anywhere where `item[CONSTANTS.IS_DIRTY]` is used without referencing one of its attributes individually.  Is there some reason to use an object/mapping for this?  I think it would make life easier if we maintained `serverDelete` and `datasetName` separately from each other.

If you're going to stick with this approach, then you should consider recasting this as:
```
    item[CONSTANTS.IS_DIRTY] = {
      [CONSTANTS.SERVER_DELETION_KEY]: false,
      [CONSTANTS.DATASET_NAME_KEY]: false,
    };
```

Reply:
> I think it would make life easier if we maintained `serverDelete` and `datasetName` separately from each other.

You mean, something like `serverDeleteIsDirty` and `datasetNameIsDirty` without a shared `isDirty`?? Yeah, maybe.

Reply:
If there are few more editable fields, those many dirty properties needed to be added. So, thought of have it in object

Reply:
In principle, I support the idea of grouping the properties together.  However, looking at the way these flags are actually used in the code, the flags are never (re-)initialized, copied, or referred to as a group -- they are always read and set individually.  And, the grouping mechanism makes referencing each of the flags more cumbersome and complex.

So, really, except in the abstract, the flags don't form a group, so there's no value in using a grouping structure in the code.

Reply:
Got different dirty flags :smile:

Reply:
Now that this code is in better shape, the places where you are managing the dirty flags as a group are becoming more obvious.  As this code continues to evolve, I think it will make sense to group these flags into a sub-object, because it will make sense to manipulate all of them at the same time.  E.g., we should have a function which checks if _any_ of them are set, by iterating over the group (and not referencing any individually); likewise, we should have a function which resets all the flags in the same way.  And, if we do it correctly, the only time(s) we will deal with a flag individually we can reference it generically via a key.  So, we'll never be accessing the flags by their "whole names", and it won't matter that the chain of references is long.